src/Symfony/Component/Messenger/Bridge/Doctrine/Tests/Transport/DoctrineReceiverTest.php

```diff
@@ -128,6 +128,188 @@ public function testFind()
         $this->assertEquals(new DummyMessage('Hi'), $actualEnvelope->getMessage());
     }
 
+    public function testAck()
+    {
+        $serializer = $this->createSerializer();
+        $connection = $this->createMock(Connection::class);
+
+        $envelope = new Envelope(new \stdClass(), [new DoctrineReceivedStamp('1')]);
+        $receiver = new DoctrineReceiver($connection, $serializer);
+
+        $connection
+            ->expects($this->once())
+            ->method('ack')
+            ->with('1')
+            ->willReturn(true);
+
+        $receiver->ack($envelope);
+    }
+
+    public function testAckThrowsRetryableException()
+    {
+        $serializer = $this->createSerializer();
+        $connection = $this->createMock(Connection::class);
+
+        $envelope = new Envelope(new \stdClass(), [new DoctrineReceivedStamp('1')]);
+        $receiver = new DoctrineReceiver($connection, $serializer);
+
+        $driverException = class_exists(Exception::class) ? Exception::new(new \PDOException('Deadlock', 40001)) : new PDOException(new \PDOException('Deadlock', 40001));
+        if (!class_exists(Version::class)) {
+            // This is doctrine/dbal 3.x
+            $deadlockException = new DeadlockException($driverException, null);
+        } else {
+            $deadlockException = new DeadlockException('Deadlock', $driverException);
+        }
+
+        $connection
+            ->expects($this->exactly(2))
+            ->method('ack')
+            ->with('1')
+            ->willReturnOnConsecutiveCalls(
```

Comment:
:boom: https://github.com/sebastianbergmann/phpunit/issues/5425

Reply:
But the deprecation has been reversed, hasn't it?

Reply:
Really!? @sebastianbergmann maybe a comment on https://github.com/sebastianbergmann/phpunit/issues/5425 could be useful?

Reply:
https://github.com/sebastianbergmann/phpunit/issues/5425#issuecomment-1996588577